.github/workflows/build.yml

```diff
@@ -0,0 +1,90 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '[0-9][0-9][0-9][0-9].[0-9]{1,2}.[0-9]+'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  py310:
+    runs-on: ubuntu-latest
+    container:
+      image: cimg/python:3.10
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run tox job
+        run: |
+          pip install tox
+          tox -e py310
+      - name: Submit code coverage data
+        run: |
+          curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import
+          curl -Os https://uploader.codecov.io/latest/linux/codecov
+          curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM
+          curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM.sig
+          gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
+          shasum -a 256 -c codecov.SHA256SUM
+          chmod +x codecov
+          ./codecov -F "$(basename $PWD | sed s/[^a-z]/_/g)"
+
+  lint:
+    runs-on: ubuntu-latest
+    container:
+      image: cimg/python:3.10
```

Comment:
Is it better to run these jobs in a python container vs doing the `actions/setup-python` step like in metric-hub? https://github.com/mozilla/metric-hub/blob/main/.github/workflows/build-docs.yml#L16-L20

Reply:
Ok, I found the [docs](https://docs.github.com/en/actions/tutorials/build-and-test-code/python#using-the-default-python-version) on this and they recommend using the `setup-python` action instead of the python container so that everything is explicit. Probably doesn't make a huge difference, but we also get caching there so maybe worth changing?